Add alarms for UIs being unavailable.

adunkman opened the pull request:
To achieve part of #250, this will notify SNS when the public UI or authenticated UI are unavailable. Subscriptions for alertees can be manually managed through the AWS console. 

- [Ran the account-specific step](https://ustaxcourt.slack.com/archives/C015SEE9CUU/p1603749691147000?thread_ts=1603749166.145600&cid=C015SEE9CUU) in the `staging` AWS account already to test. 
- Kicked off a deploy to `develop` to test in that environment, but I think the `develop` environment hasn’t been fixed since blue/green deployments were introduced. I’ll take a look in the morning, but here’s the diff for review in the meantime!

adunkman committed:
Add alarms for UIs being unavailable.

Will notifiy SNS when the public UI or authenticated UI are unavailable.

julialeague reviewed (approved):
![](https://media0.giphy.com/media/h6GTAwPQiYuVW/giphy.gif)

adunkman commented:
🎵  [Du dud dud udu duuu dudud du dudu dudu dud uduu](https://www.youtube.com/watch?v=JwZwkk7q25I) 🎵 

CircleCI [caught that we need to add permissions to the deployer](https://app.circleci.com/pipelines/github/ustaxcourt/ef-cms/1182/workflows/f4a3c48c-2be6-4689-b132-d042312e08e9/jobs/11244/parallel-runs/0/steps/0-108) for `route53:CreateHealthCheck` — following up with a commit here to adjust those permissions.

adunkman committed:
Add permissions for CI user for health checks

Route53 health checks are used to monitor the system; this PR allows the
CircleCI deployer to add, remove, modify, list them.

julialeague reviewed (approved):
"the system is down..."

adunkman committed:
Cloudwatch on AWS namespace cannot have period <60

TIL!

adunkman committed:
Use URLs to describe UI endpoints.

The previous descriptions were not helpful across environments.

adunkman commented:
Okay, finally got this sorted out — the timeframe for alarms must be 60 seconds or longer for AWS metrics, and I tweaked the alarm name after seeing them in the console, as the alarm dashboard didn’t indicate which environment the alarms are attached to.

This is ready for final review/merge.

julialeague reviewed (approved):
@mmarcotte, I recommend you accept this PR into the court environments. Thanks!
